hazelcast/src/main/java/com/hazelcast/partition/impl/InternalPartitionServiceState.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.partition.impl;
+
+/*
+    This enum represents internal state of the com.hazelcast.partition.impl.InternalPartitionServiceImpl
+    @see com.hazelcast.partition.impl.InternalPartitionServiceImpl#getMemberState
+ */
+public enum InternalPartitionServiceState {
+
+    SAFE,
```

Comment:
Can you add documentation what the meaning is of each of these states?

And is it the state of the InternalPartitionService, or is it the state of the partition?

Perhaps call it PartitionMigrationStatus?